giturlparse/result.py

```diff
@@ -51,11 +53,28 @@ def user(self):
 
         return self.owner
 
+    @property
+    def groups(self):
+        if self.groups_path:
+            return self.groups_path.split('/')
+        else:
+            return []
+
+    @property
+    def path(self):
+        if self.raw_path:
+            return self.raw_path.replace('/blob/', '').strip('/')
```

Comment:
I dont see the benefit of removing only `blob` (and not other REST 'commands' like 'tree'), but not also remove the branch name.  Keeping the branch name in the `path` makes it like SVN, but for most VCS the branch and the rest of the path are separate and are used in different commands/parts of the workflow.  If they are combined here, people will usually need to split it before they can use it.

If the branch name is also removed, `path` becomes synonymous with the file path that is found in a local clone.

Exposing the `branch` as a separate attribute would be useful, but can be done later.

fwiw, git-url-parse uses `pathname` for what you have exposed as `raw_path`, to correspond with urlparse()'s `pathname`.  I dont mind about this, as a compatibility shim can do simple mapping like this when the two libraries are close enough that a shim is sufficient.

Reply:
Regarding the capturing group name, I don't see any reason not to match a more sensible name, I will change that

Regarding the other rest commands, we only need to detail them, it makes sense to  remove all of them if course

Regarding the branch name I've not come up with a working regexp to tell the branch name from the file path, as the branch name can contain slashes, more than happy to strip the branch from path and expose it as a variable

Reply:
`pathname` is now the same as `git-url-parse`, but I could not find a way to get the branch for the `blob` verb

`tree` verb instead can return the branch name